Realm/RLMUtil.mm

```diff
@@ -196,12 +196,16 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
 NSDictionary *RLMValidatedDictionaryForObjectSchema(id value, RLMObjectSchema *objectSchema, RLMSchema *schema, bool allowMissing) {
     NSArray *properties = objectSchema.properties;
     NSMutableDictionary *outDict = [NSMutableDictionary dictionaryWithCapacity:properties.count];
+    NSDictionary *defaultValues = nil;
     for (RLMProperty *prop in properties) {
         id obj = [value valueForKey:prop.name];
 
         // get default for nil object
         if (!obj && !allowMissing) {
-            obj = objectSchema.defaultValues[prop.name];
+            if (!defaultValues) {
+                defaultValues = RLMDefaultValuesForObjectSchema(objectSchema);
```

Comment:
`RLMDefaultValuesForObjectSchema` needs to not be so slow if it's going to be called for every object that needs a default value. At the minimum it needs to not make a mutable copy of the dictionary for obj-c objects, and the whole thing should probably be refactored to just grab the values from the swift object directly rather than converting the object to a dictionary.